docs/spec/slashing/state-machine.md

```diff
@@ -0,0 +1,174 @@
+if you misconfigure your HSM and double-sign a bunch of old blocks, you'll only be punished for the first double-sign (and then immediately jailed,
+so that you have a chance to reconfigure your setup). This will still be quite expensive and desirable to avoid, but slashing periods somewhat blunt
+the economic impact of unintentional misconfiguration.
+
+A new slashing period starts whenever a validator is bonded and ends whenever the validator is unbonded (which will happen if the validator is jailed).
+The amount of tokens slashed relative to validator power for infractions committed within the slashing period, whenever they are discovered, is capped
+at the punishment for the worst infraction (which for the Cosmos Hub at launch will be double-signing a block).
+
+#### ASCII timelines
+
+*Code*
+
+*[*   : timeline start  
+*]*   : timeline end  
+*<*   : slashing period start  
+*>*   : slashing period end  
+*C<sub>n</sub>* : infraction `n` committed  
+*D<sub>n</sub>* : infraction `n` discovered  
+*V<sub>b</sub>* : validator bonded  
+*V<sub>u</sub>* : validator unbonded  
+
+*Single infraction*
+
+<----------------->   
+[----------C<sub>1</sub>----D<sub>1</sub>,V<sub>u</sub>-----]
+
+A single infraction is committed then later discovered, at which point the validator is unbonded and slashed at the full amount for the infraction.
+
+*Multiple infractions*
+
+<---------------------------->   
+[----------C<sub>1</sub>--C<sub>2</sub>---C<sub>3</sub>---D<sub>1</sub>,D<sub>2</sub>,D<sub>3</sub>V<sub>u</sub>-----]
+
+Multiple infractions are committed within a single slashing period then later discovered, at which point the validator is unbonded and slashed for only the worst infraction.
+
+*Multiple infractions after rebonding*
+
+
+<---------------------------->&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<-------------->  
+[----------C<sub>1</sub>--C<sub>2</sub>---C<sub>3</sub>---D<sub>1</sub>,D<sub>2</sub>,D<sub>3</sub>V<sub>u</sub>---V<sub>b</sub>---C<sub>4</sub>----D<sub>4</sub>,V<sub>u</sub>--]
+
+Multiple infractions are committed within a single slashing period then later discovered, at which point the validator is unbonded and slashed for only the worst infraction.
+The validator then unjails themself and rebonds, then commits a fourth infraction - which is discovered and punished at the full amount, since a new slashing period started
+when they unjailed and rebonded.
+
+## Transactions
+
+In this section we describe the processing of transactions for the `slashing` module.
+
+### TxUnjail
+
+If a validator was automatically unbonded due to downtime and wishes to come back online &
+possibly rejoin the bonded set, it must send `TxUnjail`:
+
+```
+type TxUnjail struct {
+    ValidatorAddr sdk.AccAddress
+}
+
+handleMsgUnjail(tx TxUnjail)
+
+    validator = getValidator(tx.ValidatorAddr)
+    if validator is nil
```

Comment:
I'd rather still see `if validator == nil` myself - either way we should stay consistent throughout the spec, what I've suggested is what's currently in fee's / staking spec I believe

Reply:
Hmm, I'm not sure I agree, I think all-English pseudocode might be better (see @alexanderbez comments above). Changed for consistency for now, this is a "spec spec" issue.

Reply:
Okay - this should be agreed upon and used in the _spec for specs_ - agree to disagree for this PR I think

Reply:
(I changed to `==` for now anyways)